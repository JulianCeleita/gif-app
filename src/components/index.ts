
// Archivos de barril, * = Exporte todo de tal componente.

export * from './AddCategory';
export * from './GifGrid';
export * from './GifItem';
