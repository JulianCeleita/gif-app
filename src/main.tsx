import React from 'react'
import ReactDOM from 'react-dom/client'
import { GifApp } from './GifApp';
import './styles.css'

ReactDOM.createRoot(document.getElementById('root') as HTMLElement).render(
  <React.StrictMode>                        {/* Solo se usa en modo desarrollo, para crear warnings y avisos de APIs, no afecta la producci√≥n */}
    <GifApp />
  </React.StrictMode>                       // Lo desactivo para evitar que es escriba duplicado en consola
)
