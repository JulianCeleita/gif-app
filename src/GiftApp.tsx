import { useState } from "react";
import { AddCategory } from './components/AddCategory';


export const GifApp = () => {

    const [category, setCategory] = useState([ 'One Punch', 'Dragón' ]);      // Este arreglo será llamado por el .map para que aparezca
    
    const onAddCategory = ( ) => {                      // Para agregar una categoría de gits

        setCategory([ ...category, 'Valorant' ]);        //Toma la ultima impresión de category y le agrega el elemento valorant
    };

    return (
      <>
          {/* Title */}
          <h1>Gif App</h1>
  
          {/* Input */}
          <AddCategory setCategory={setCategory} />
  
          {/* Listado de Gifs */}
          <ol>
            { category.map( category => {                          // Barre cada uno de los elementos del arreglo y les da el valor de abajo
                return <li key={category}> {category} </li>      // Debo darle un key unico, como su identificador
            })}
          </ol>

            {/* Gift items */}
      </>
    )
  }