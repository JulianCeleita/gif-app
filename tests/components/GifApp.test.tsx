import React from 'react';
import {describe, expect, test} from '@jest/globals';
import { render } from '@testing-library/react';
import { GifApp } from '../../src/GifApp';

describe ('Test in <GifApp/>', ()=> {

    test('Should match the snapshot', () => {

        const { container } = render( <GifApp/> );
        expect( container ).toMatchSnapshot();

    });
})