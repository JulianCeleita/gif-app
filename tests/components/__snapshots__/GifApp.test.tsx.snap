// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <GifApp/> Should match the snapshot 1`] = `
<div>
  <h1>
    Gif App
  </h1>
  <form
    aria-label="form"
  >
              
    <input
      placeholder="Buscar Gifs"
      type="text"
      value=""
    />
  </form>
                
  <h3>
     
    Ferrari
     
  </h3>
  <h2>
     Loading... 
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
